### PR TITLE
feat(plausible)!: Migrate to Arrow native SDK

### LINKED ISSUE #10762: feat: Migrate `plugin/source/plausible` to `github.com/cloudquery/plugin-sdk/v3`
### Which problem is this feature request solving?

Part of #10711

### Describe the solution you'd like

Upgrade the code.

### Pull request (optional)

- [X] I can submit a pull request

### PR DESCRIPTION
**Pull request by disq**
Closes #10762

BEGIN_COMMIT_OVERRIDE
feat: Update to use [Apache Arrow](https://arrow.apache.org/) type system (#11003)

BREAKING-CHANGE: This release introduces an internal change to our type system to use [Apache Arrow](https://arrow.apache.org/). This should not have any visible breaking changes, however due to the size of the change we are introducing it under a major version bump to communicate that it might have some bugs that we weren't able to catch during our internal tests. If you encounter an issue during the upgrade, please submit a [bug report](https://github.com/cloudquery/cloudquery/issues/new/choose). You will also need to update destinations depending on which one you use:
- Azure Blob Storage >= v3.2.0
- BigQuery >= v3.0.0
- ClickHouse >= v3.1.1
- DuckDB >= v1.1.6
- Elasticsearch >= v2.0.0
- File >= v3.2.0
- Firehose >= v2.0.2
- GCS >= v3.2.0
- Gremlin >= v2.1.10
- Kafka >= v3.0.1
- Meilisearch >= v2.0.1
- Microsoft SQL Server >= v4.2.0
- MongoDB >= v2.0.1
- MySQL >= v2.0.2
- Neo4j >= v3.0.0
- PostgreSQL >= v4.2.0
- S3 >= v4.4.0
- Snowflake >= v2.1.1
- SQLite >= v2.2.0

END_COMMIT_OVERRIDE

### CODE REVIEW

File: plugins/source/plausible/resources/services/stats_timeseries.go [hunk 1]
```diff
@@ -19,45 +20,41 @@ func StatsTimeseries() *schema.Table {
...
 			},
 			{
 				Name:        "date",
 				Description: "Date of the data point",
-				Type:        schema.TypeTimestamp,
-				CreationOptions: schema.ColumnCreationOptions{
-					PrimaryKey: true,
-				},
-				Resolver: ResolveDate,
+				Type:        arrow.FixedWidthTypes.Timestamp_s,
```
Comment (disq) [inline plugins/source/plausible/resources/services/stats_timeseries.go:30]: This is different than our usual `Timestamp_us` approach. Initially wanted to use `Date64` but our scalar pkg doesn't support it yet.

Comment (hermanschaaf) [reply plugins/source/plausible/resources/services/stats_timeseries.go:30]: Why can't we use `Timestamp_us`?

Comment (disq) [reply plugins/source/plausible/resources/services/stats_timeseries.go:30]: we can, but it's a date in the end. Would've used Date64 if scalar was up to date

Comment (hermanschaaf) [reply plugins/source/plausible/resources/services/stats_timeseries.go:30]: Even though it's a date, we don't yet have a compatibility mode, so we should try and stick to the Arrow types that we determined to be the equivalents of the old CQTypes for now (in this case `Timestamp_us`). This should guarantee that the schema doesn't break for the majority of destinations, if not all.

Comment (hermanschaaf) [reply plugins/source/plausible/resources/services/stats_timeseries.go:30]: (More context: this is what the struct transformer is doing now https://github.com/cloudquery/plugin-sdk/blob/main/transformers/struct.go#L307C6-L350 so if we require manual column updates we should also map it in the same way)

Comment (disq) [reply plugins/source/plausible/resources/services/stats_timeseries.go]: ok fixed :)

**Review by candiduslynx — APPROVED** [review]

**Review by hermanschaaf — APPROVED** [review]

**Pull request merged**